libs/auth-api-lib/src/lib/delegations/delegations-index.service.ts

```diff
@@ -170,6 +175,45 @@ export class DelegationsIndexService {
     await this.saveToIndex(delegations)
   }
 
+  /* Add item to index */
+  async createOrUpdateDelegationRecord(delegation: DelegationRecordInputDTO) {
+    const valid = validateDelegationTypeAndProvider(
+      delegation.type,
+      delegation.provider,
+    )
+
+    if (!valid) {
+      throw new Error('Invalid delegation type and provider combination')
+    }
+
```

Comment:
We should also verify that the nationalIds are valid and not the same value and that the `validTo` field, if provided, needs to be in the future.